beacon/engine/types.go

```diff
@@ -32,15 +32,25 @@ import (
 // PayloadAttributes describes the environment context in which a block should
 // be built.
 type PayloadAttributes struct {
-	Timestamp             uint64              `json:"timestamp"             gencodec:"required"`
-	Random                common.Hash         `json:"prevRandao"            gencodec:"required"`
-	SuggestedFeeRecipient common.Address      `json:"suggestedFeeRecipient" gencodec:"required"`
-	Withdrawals           []*types.Withdrawal `json:"withdrawals"`
+	Timestamp             uint64         `json:"timestamp"             gencodec:"required"`
+	Random                common.Hash    `json:"prevRandao"            gencodec:"required"`
+	SuggestedFeeRecipient common.Address `json:"suggestedFeeRecipient" gencodec:"required"`
+	// <specular modification>
+	Withdrawals []*types.Withdrawal `json:"withdrawals,omitempty" gencodec:"optional"`
+	// Transactions is a field for rollups: the transactions list is forced into the block
+	Transactions [][]byte `json:"transactions,omitempty"  gencodec:"optional"`
+	// NoTxPool is a field for rollups: if true, the no transactions are taken out of the tx-pool,
+	// only transactions from the above Transactions list will be included.
+	NoTxPool bool `json:"noTxPool,omitempty" gencodec:"optional"`
```

Comment:
is there any place where the decision to not to include a gas-limit param is documented / explained?
i'm only aware of this https://github.com/ethereum/builder-specs/issues/29 which generally seems to favour including it

Reply:
Discussed on call this morning:

We're going to investigate if we should make this change, and what approach to take. But we can do it in a separate PR.